Archive/Wiedijk100Theorems/Partition.lean

```diff
@@ -130,7 +130,9 @@ theorem num_series' [Field α] (i : ℕ) :
       simp only [coeff_one, if_false, mul_sub, mul_one, coeff_indicator,
         LinearMap.map_sub, reduceCtorEq]
       simp_rw [coeff_mul, coeff_X_pow, coeff_indicator, @boole_mul _ _ _ _]
-      erw [sum_ite, sum_ite]
+      -- TODO: why is `erw` needed here?
+      erw [sum_ite]
+      rw [sum_ite]
```

Comment:
This works:
```suggestion
      rw [sum_ite (hp := fun _ ↦ Classical.propDecidable _), sum_ite]
```
Without explicitly giving `hp`, `rw` is synthesising `Set.decidableSetOf x.1 (Dvd.dvd (i + 1))` for the `hp` field rather than the goal's `Classical.propDecidable …`.

Reply:
Thanks!